frontend/src/scenes/billing/BillingHero.tsx

```diff
@@ -1,26 +1,229 @@
+import { billingLogic } from './billingLogic'
+import { billingProductLogic } from './billingProductLogic'
+import { paymentEntryLogic } from './paymentEntryLogic'
+import { PlanComparisonModal } from './PlanComparison'
 
-    return (
-        <div className="BillingHero" ref={billingHeroRef}>
-            <div className="p-4">
-                <p className="text-xs uppercase my-0">How pricing works</p>
-                <h1 className="ingestion-title">Get the whole hog.</h1>
-                <h1 className="ingestion-title text-danger">Only pay for what you use.</h1>
-                <p className="mt-2 mb-0">
-                    Upgrade to get access to premium product and platform features. Set billing limits as low as $0 to
-                    control spend.
+const PLAN_BADGES: Record<BillingPlan, string> = {
+    [BillingPlan.Free]: planFree,
+    [BillingPlan.Paid]: planPaid,
+    // TODO: Add teams badge once ready
+    [BillingPlan.Teams]: planPaid,
+    [BillingPlan.Enterprise]: planEnterprise,
+}
+
+const STARTUP_PROGRAM_BADGES: Record<StartupProgramLabel, string> = {
+    [StartupProgramLabel.YC]: planYc,
+    [StartupProgramLabel.Startup]: planStartup,
+}
+
+interface CopyVariation {
+    title: string | null
+    subtitle: string | null
+    getDescription: (billingPlan: BillingPlan, scrollToProduct: (productType: string) => void) => JSX.Element
+    backgroundColor: string
+}
+
+const BADGE_CONFIG: Record<BillingPlan | StartupProgramLabel, CopyVariation> = {
+    [BillingPlan.Free]: {
+        title: 'Get the whole hog.',
+        subtitle: 'Only pay for what you use.',
+        backgroundColor: 'bg-danger-highlight',
+        getDescription: () => (
+            <>
+                <p>Add your credit card to remove usage limits and unlock all platform features.</p>
+                <p>Set billing limits as low as $0 to control your spend.</p>
+                <p className="italic">P.S. You still keep the monthly free allowance for every product!</p>
+            </>
+        ),
+    },
+    [BillingPlan.Paid]: {
+        title: 'Good call!',
+        subtitle: "You're on the Ridiculously Cheap™ plan.",
+        backgroundColor: 'bg-warning-highlight',
+        getDescription: (_billingPlan: BillingPlan, scrollToProduct: (productType: string) => void) => (
+            <p>
+                If you're growing like crazy, you might want to check out the{' '}
+                {scrollToProduct ? (
+                    <>
+                        <Link onClick={() => scrollToProduct('teams')}>Teams</Link>
+                        {' or '}
+                        <Link onClick={() => scrollToProduct('enterprise')}>Enterprise</Link>
+                    </>
+                ) : (
+                    'Teams or Enterprise'
+                )}{' '}
+                plan.
+            </p>
+        ),
+    },
+    [BillingPlan.Teams]: {
+        title: 'Good call!',
+        subtitle: "You're on the Teams plan.",
```

Comment:
Can we update this to say `You're on the Ridiculously Cheap™ plan (with Teams)` or something like that until we have the badge, it's a big confusing.